Upgrade interactors to v0.28.0

cowboyd opened the pull request:
Recently, a more flexible mechanism for defining interactors was added to `@bigtest/interactor` including the ability re-use and extend interactors via the "builder" syntax. Among other things, this means we can easily inherit the shared filters and actions of all html elements such as `id`, `title`, `click()`, and `focus()`.

This upgrades to the new version of the library, and refactors the interactors away from the raw `createInteractor` syntax.

- [x] re-export `HTML` interactor and builtin matchers
- [x] accordion
- [x] button
- [x] checkbox
- [x] converge
- [x] dropdown
- [x] icon-button
- [x] index
- [x] key-value
- [x] keyboard
- [x] label
- [x] layer
- [x] multi-column-list
- [x] pane
- [x] radio-button
- [x] search-field
- [x] select
- [x] text-field
- [x] textarea
- [x] tooltip

elrickvm commented:
Howdy, cowboy. I have push access now. Woot Woot! I was able to refactor the remaining Interactors to the new Builder syntax and the branch should be updated now.

elrickvm commented:
@cowboyd I updated the Interactor version to `v0.28.0` and updated the branch name to correspond to the bump version.

cowboyd commented:
> @cowboyd I updated the Interactor version to `v0.28.0` and updated the branch name to correspond to the bump version.
Nice @elrickvm! Were there any incompatibilities between .26 and .28?

cowboyd commented on interactors/tooltip.js:
```diff
@@ -15,16 +15,14 @@ export const Tooltip = createInteractor('tooltip')({
+  ])
+  .filters({
     text: (el) => el.querySelector('[class^=text]').textContent,
     id: (el) => el.id,
```
`id` is now builtin 🛠️

id-jenkins commented:
<html><body><pre>
yarn run v1.22.4
$ eslint .

/home/jenkins/workspace/folio-org_stripes-testing_PR-100/project/interactors/dropdown.js
  19:73  error  'isVisible' is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)

info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
</pre><body></html>

cowboyd commented on interactors/dropdown.js:
```diff
@@ -1,20 +1,14 @@
+const DropdownMenu = HTML.extend('dropdown menu')
+  .selector('div[class*=DropdownMenu]')
+  .filters({
+  });
```
It looks like there's not any more need for the filters at all?

cowboyd commented on interactors/keyboard.js:
```diff
@@ -1,54 +1,53 @@
     altLeft: press('AltLeft')
-  }
-})();
+  })();
```
In other words @elrickvm , I think we can just remove the `()` from the end? Try if this works.
```suggestion
  });
```

cowboyd commented on interactors/multi-column-list.js:
```diff
@@ -1,44 +1,42 @@
-});
+  })
+  .actions({
+    click: ({ perform }) => perform((el) => el.click())
```
@elrickvm I think we can remove the actions hash completely now since the only one action, `click()` and that is provided by the `HTML` interactor.

cowboyd commented on interactors/pane.js:
```diff
@@ -30,13 +29,11 @@ export default createInteractor('pane')({
+  })
+  .actions({
     dismiss: (interactor) => interactor.find(Button({ ariaLabel: 'Close ' })).click(),
     // TODO, enable these when updating to >0.22.0
     // we should also add some tests for these?
     // focus,
     // blur
```
@elrickvm We can now remove these dead comments, right?
